ingest/muxer.go

```diff
@@ -882,31 +889,85 @@ func (im *IngestMuxer) SyncContext(ctx context.Context, to time.Duration) error
 			break
 		}
 		if im.connHot == 0 {
-			im.mtx.Unlock()
 			return ErrAllConnsDown
 		}
 		//only check for a timeout if to is greater than zero.  A zero value or negative value means no timeout
 		if to > 0 && time.Since(ts) > to {
-			im.mtx.Unlock()
 			return ErrTimeout
 		}
 	}
 
+	timeLeft := to - time.Since(ts)
+	if timeLeft <= 0 {
+		return ErrTimeout
+	}
+
+	//check for the simple case of a single indexer
+	if len(im.igst) == 1 {
+		var err error
+		if ig := im.igst[0]; ig != nil {
+			err = ig.syncTimeout(timeLeft)
+		} else {
+			err = ErrAllConnsDown
+		}
+		return err
+	}
+
+	//DO NOT CLOSE the channel unless we get all of them back
+	total := len(im.igst)
+	ch := make(chan error, total)
+	tmr := time.NewTimer(timeLeft + time.Second) //there will be some slop here
+	defer tmr.Stop()
+
+	// do a parallel sync
 	var count int
 	for _, v := range im.igst {
-		if v != nil {
-			if err := v.Sync(); err != nil {
-				if err == ErrNotRunning {
-					count++
+		go func(ig *IngestConnection, ech chan error) {
+			if ig == nil {
+				ech <- nil
+			} else {
+				ech <- ig.syncTimeout(timeLeft)
+			}
+		}(v, ch)
+		count++
+	}
+
+	//now go read them all
+	var good int
+	var down int
+	var timeout bool
+	var lastErr error
+loop:
+	for count > 0 {
+		select {
+		case err := <-ch:
+			count--
+			if err == nil {
+				good++
+			} else {
+				if lastErr == nil {
+					lastErr = err
```

Comment:
we only keep one error here. Might we merge them instead?

Reply:
added merge utility, please validate that i got the logic right.